agent/connect/generate.go

```diff
@@ -35,6 +40,11 @@ func pemEncode(value []byte, blockType string) (string, error) {
 func generateRSAKey(keyBits int) (crypto.Signer, string, error) {
 	var pk *rsa.PrivateKey
 
+	// Check for a secure key length.
+	if keyBits < MinPrivateKeyBitsRSA {
+		return nil, "", fmt.Errorf("error generating RSA private key: key size must be at least %d bits", MinPrivateKeyBitsRSA)
```

Comment:
UX: would be nice to surface the `"got: %d", keyBits` as part of the error message